plugins/outputs/stan/stan.go

```diff
@@ -0,0 +1,152 @@
+package stan
+
+import (
+	"fmt"
+
+	nats_client "github.com/nats-io/go-nats"
+	stan_client "github.com/nats-io/go-nats-streaming"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/outputs"
+	"github.com/influxdata/telegraf/plugins/serializers"
+)
+
+type STAN struct {
+	// NATS Streaming Cluster ID
+	ClusterID string `toml:"cluster_id"`
+
+	// NATS Streaming Client ID
+	ClientID string `toml:"client_id"`
+
+	// Servers is the NATS server pool to connect to
+	Servers []string
+	// Credentials
+	Username string
+	Password string
+
+	// STAN subject to publish metrics to
+	Subject string
+
+	// Path to CA file
+	SSLCA string `toml:"ssl_ca"`
+	// Path to host cert file
+	SSLCert string `toml:"ssl_cert"`
+	// Path to cert key file
+	SSLKey string `toml:"ssl_key"`
+	// Use SSL but skip chain & host verification
+	InsecureSkipVerify bool
+
+	conn       stan_client.Conn
+	serializer serializers.Serializer
+}
+
+var sampleConfig = `
+  ## NATS Streaming Cluster ID
+  cluster_id = "test-cluster"
+  ## Client ID
+  client_id = "telegraf-client-id"
+  ## URLs of NATS servers
+  servers = ["nats://localhost:4222"]
+  ## Optional credentials
+  # username = ""
+  # password = ""
+  ## NATS subject for producer messages
+  subject = "telegraf"
+
+  ## Optional SSL Config
+  # ssl_ca = "/etc/telegraf/ca.pem"
+  # ssl_cert = "/etc/telegraf/cert.pem"
+  # ssl_key = "/etc/telegraf/key.pem"
+  ## Use SSL but skip chain & host verification
+  # insecure_skip_verify = false
+
+  ## Data format to output.
+  ## Each data format has its own unique set of configuration options, read
+  ## more about them here:
+  ## https://github.com/influxdata/telegraf/blob/master/docs/DATA_FORMATS_OUTPUT.md
+  data_format = "influx"
+`
+
+func (n *STAN) SetSerializer(serializer serializers.Serializer) {
+	n.serializer = serializer
+}
+
+func (n *STAN) Connect() error {
+	var err error
+
+	natsOpts := nats_client.DefaultOptions
+
+	// override max reconnection tries
+	natsOpts.MaxReconnect = -1
+
+	// override servers, if any were specified
+	natsOpts.Servers = n.Servers
+
+	// override authentication, if any was specified
+	if n.Username != "" {
+		natsOpts.User = n.Username
+		natsOpts.Password = n.Password
+	}
+
+	// override TLS, if it was specified
+	tlsConfig, err := internal.GetTLSConfig(
+		n.SSLCert, n.SSLKey, n.SSLCA, n.InsecureSkipVerify)
+	if err != nil {
+		return err
+	}
+	if tlsConfig != nil {
+		// set NATS connection TLS options
+		natsOpts.Secure = true
+		natsOpts.TLSConfig = tlsConfig
+	}
+
+	// try and connect to the NATS server
+	natsConn, err := natsOpts.Connect()
+	if err != nil {
+		return err
+	}
+
+	// try and connect to the NATS Streaming cluster
+	n.conn, err = stan_client.Connect(n.ClusterID, n.ClientID, stan_client.NatsConn(natsConn))
```

Comment:
If you are going to break arguments by line above, I suggest doing the same here for consistency.